src/logic/hooks/useCheckIfTransactionWillFail.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react'
+import { checkIfExecTxWillFail } from 'src/logic/safe/transactions/gas'
+
+type TransactionWillFailProps = {
+  data: string
+  safeAddress: string
+  txRecipient: string
+  txAmount?: string
+}
+
+export const useCheckIfTransactionWillFail = ({
+  data,
+  safeAddress,
+  txAmount,
+  txRecipient,
+}: TransactionWillFailProps): boolean => {
```

Comment:
Agus, can we add a loading status here? 
perhaps not now, but in the future, we might want to show the user a loading status and avoid flickers.

Reply:
Mm not sure about what do you mean, you mean, adding a `loading` prop? how should I use that?

Reply:
a possibility could be status: loading | success | error